packages/client/src/rtc/Dispatcher.ts

```diff
@@ -31,41 +74,43 @@ export const isSfuEvent = (
   return Object.prototype.hasOwnProperty.call(sfuEventKinds, eventName);
 };
 
-export type SfuEventListener = (event: SfuEvent) => void;
-
 export class Dispatcher {
-  private subscribers: {
-    [eventName: string]: SfuEventListener[] | undefined;
-  } = {};
-  private readonly logger: Logger = getLogger(['sfu-client']);
+  private readonly logger = getLogger(['Dispatcher']);
+  private subscribers: Partial<
+    Record<SfuEventKinds, CallEventListener<any>[] | undefined>
+  > = {};
 
   dispatch = (message: SfuEvent) => {
     const eventKind = message.eventPayload.oneofKind;
     if (eventKind) {
-      this.logger(
-        'debug',
-        `Dispatching ${eventKind}`,
-        (message.eventPayload as any)[eventKind],
-      );
+      const payload = (message.eventPayload as any)[eventKind];
```

Comment:
To prevent explicit casting here, you can use a helper type:

```ts
type DispatchableMessage<K extends SfuEventKinds> = {
  eventPayload: {
    oneofKind: K;
  } & {
    [Key in K]: AllSfuEvents[Key];
  };
};
```

And use it like this:

```ts
dispatch = <K extends SfuEventKinds>(message: DispatchableMessage<K>) => {
  /* ... */
}
```

Reply:
hey @myandrienko, can you check this one? https://github.com/GetStream/stream-video-js/pull/1246/commits/37caaa477cba7392906d53d467ddf8d65c25e58e